docs/aws/audit/cloudtrailmonitoring/rules/cloudtrail_object_lock_s3_buckets_remediation.mdx

```diff
@@ -1,6 +1,266 @@
+3. **Enable Object Lock on S3 Bucket**: Use a Python script to enable Object Lock on the S3 bucket.
+4. **Integrate with CloudTrail**: Ensure CloudTrail is configured to use the S3 bucket with Object Lock enabled.
+
+By following these steps, you can prevent the misconfiguration where the Object Lock feature should be enabled in CloudTrail using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the CloudTrail console at https://console.aws.amazon.com/cloudtrail/.
+
+2. In the navigation pane, click on "Trails".
+
+3. In the Trails list, select the trail for which you want to check the Object Lock feature.
+
+4. In the details pane, under the "S3 bucket" section, click on the link of the S3 bucket that is associated with the selected trail.
+
+5. In the S3 bucket page, click on the "Properties" tab. Under the "Object Lock" section, you can check if the Object Lock feature is enabled or not. If it is enabled, it will show "Enabled". If it is not enabled, it will show "Disabled".
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can do this by following the instructions provided by AWS. Make sure you have the necessary permissions to access the CloudTrail service.
+
+2. Once the AWS CLI is set up, you can use the following command to list all the trails in your AWS account:
+
+   ```
+   aws cloudtrail describe-trails
+   ```
+
+   This command will return a JSON output with the details of all the trails.
+
+3. To check if the Object Lock feature is enabled for a specific trail, you can use the following command:
+
+   ```
+   aws s3api get-object-lock-configuration --bucket <bucket-name>
+   ```
+
+   Replace `<bucket-name>` with the name of the S3 bucket where the trail logs are stored. This command will return the Object Lock configuration for the specified bucket.
+
+4. If the Object Lock feature is enabled, the command will return a configuration that includes the `ObjectLockEnabled` field set to `Enabled`. If the feature is not enabled, the command will not return this field. 
+
+   Please note that you need to have the necessary permissions to get the Object Lock configuration of a bucket. If you don't have these permissions, the command will return an error.
+</Accordion>
+
+<Accordion title='Using Python'>
+To detect if the Object Lock feature is enabled in CloudTrail using Python scripts, you can use the Boto3 library, which allows you to directly interact with AWS services, including S3 where CloudTrail logs are stored. Here are the steps:
+
+1. **Import the necessary libraries and establish a session with AWS:**
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>